crates/miden-objects/src/account/storage/map/partial.rs

```diff
@@ -11,25 +13,47 @@ use miden_crypto::merkle::{
 };
 
 use crate::account::{StorageMap, StorageMapWitness};
+use crate::utils::serde::{ByteReader, DeserializationError};
 
 /// A partial representation of a [`StorageMap`], containing only proofs for a subset of the
 /// key-value pairs.
 ///
 /// A partial storage map carries only the Merkle authentication data a transaction will need.
 /// Every included entry pairs a value with its proof, letting the transaction kernel verify reads
 /// (and prepare writes) without needing the complete tree.
+///
+/// ## Guarantees
+///
+/// This type guarantees that the original key-value pairs it contains are all present in the
+/// contained partial SMT. Note that the inverse is not necessarily true. The SMT may contain more
+/// entries than the map because to prove inclusion of a given original key A an
+/// [`SmtLeaf::Multiple`] may be present that contains both keys hash(A) and hash(B). However, B may
+/// not be present in the key-value pairs and this is a valid state.
 #[derive(Clone, Debug, PartialEq, Eq, Default)]
 pub struct PartialStorageMap {
     partial_smt: PartialSmt,
+    map: BTreeMap<Word, Word>,
```

Comment:
nit: would `entries` be a better name than `map` here?

I would also maybe leave a doc comment that the `key` here is the un-hashed key.

Reply:
Sounds good! Changed to `entries` across `(Partial)StorageMap` and `StorageMapWitness` and added doc comments.